esmvaltool/cmorizers/obs/cmorize_obs_lai3g.py

```diff
@@ -0,0 +1,216 @@
+        # (Raw data has two values per month)
+        cube = cubes.concatenate_cube()
+        iris.coord_categorisation.add_month_number(cube, 'time')
+        cube = cube.aggregated_by('month_number', iris.analysis.MEAN)
+
+        # Cache cube on disk to save memory
+        cached_path = os.path.join(in_dir, f'{year}.nc')
+        iris.save(cube, cached_path)
+        logger.info("Cached %s", cached_path)
+        nc_files.append(cached_path)
+
+    # Build final cube
+    logger.info("Building final cube")
+    cubes = iris.cube.CubeList()
+    for nc_file in nc_files:
+        cube = iris.load_cube(nc_file)
+        cubes.append(cube)
+    final_cube = cubes.concatenate_cube()
+    utils.fix_var_metadata(final_cube, cmor_info)
+    utils.convert_timeunits(final_cube, 1950)
+    utils.fix_coords(final_cube)
+    if not cfg.get('regrid'):
+        utils.flip_dim_coord(final_cube, 'latitude')
+    utils.set_global_atts(final_cube, attrs)
+    utils.save_variable(final_cube,
+                        cmor_info.short_name,
+                        out_dir,
+                        attrs,
+                        unlimited_dimensions=['time'])
+
+
+def _get_coords(year, filename, cfg):
+    """Get correct coordinates for cube."""
+    time_units = Unit('days since 1950-1-1 00:00:00', calendar='standard')
+
+    # Extract date from filename
+    time_str = filename.replace(cfg['binary_prefix'], '')
+    month = MONTHS[time_str[4:7]]
+    day = DAYS[time_str[7:8]]
+    date = datetime(year, month, day)
+
+    # Build time coordinate
+    time_data = [time_units.date2num(date)]
+    time_coord = iris.coords.DimCoord(time_data,
+                                      standard_name='time',
+                                      long_name='time',
+                                      var_name='time',
+                                      units=time_units)
+
+    # Build latitude/Longitude coordinates
+    latitude_data = np.linspace(UPPER_LEFT_LAT, LOWER_RIGHT_LAT, N_LAT)
+    longitude_data = np.linspace(UPPER_LEFT_LON, LOWER_RIGHT_LON, N_LON)
+    lat_coord = iris.coords.DimCoord(latitude_data,
+                                     standard_name='latitude',
+                                     long_name='latitude',
+                                     var_name='lat',
+                                     units='degrees')
+    lon_coord = iris.coords.DimCoord(longitude_data,
+                                     standard_name='longitude',
+                                     long_name='longitude',
+                                     var_name='lon',
+                                     units='degrees')
+
+    return [(time_coord, 0), (lat_coord, 1), (lon_coord, 2)]
+
+
+def _get_years(in_dir, cfg):
+    """Get all available years from input directory."""
+    bin_files = os.listdir(in_dir)
+    bin_files = [f.replace(cfg['binary_prefix'], '') for f in bin_files]
+    years = [int(f[:4]) for f in bin_files]
```

Comment:
```suggestion
    years = {int(f[:4]) for f in bin_files}
```
and remove the list(set(..)) from the next line